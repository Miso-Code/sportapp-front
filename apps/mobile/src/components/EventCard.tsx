import React from 'react'
import dayjs from 'dayjs'
import { StyleSheet, View } from 'react-native'
import { Card, Text, Chip, useTheme, MD3Theme } from 'react-native-paper'
import { useTranslation } from 'react-i18next'

interface EventCardProps {
	date: string
	title: string
	description: string
}

const EventCard: React.FC<EventCardProps> = ({ date, title, description }) => {
	const theme = useTheme()
	const styles = createStyles(theme)
	const { t, i18n } = useTranslation()

	const parsedDate = dayjs(date).locale(i18n.language).format('DD MMMM YYYY')

	const isUpcoming = dayjs(date).isAfter(dayjs(), 'days')

	return (
		<Card elevation={1} style={styles.card}>
			<Card.Content>
				<Text variant='labelLarge' style={styles.dateLabel}>
					{parsedDate}
				</Text>
				<View style={styles.eventInfo}>
					<Text variant='titleMedium'>{title}</Text>
					<Text variant='bodyMedium'>{description}</Text>
				</View>
				<Chip
					testID='chip'
					style={isUpcoming ? styles.chipUpcoming : styles.chipToday}
					textStyle={
						isUpcoming
							? styles.chipUpcomingText
							: styles.chipTodayText
					}
					compact>
					{t(isUpcoming ? 'training.upcoming' : 'training.today')}
				</Chip>
			</Card.Content>
		</Card>
	)
}

const createStyles = (theme: MD3Theme) =>
	StyleSheet.create({
		card: {
			backgroundColor: '#f2f2f2',
			paddingBottom: 10
		},
		cardTitle: {
			fontWeight: 'bold'
		},
		dateLabel: {
			opacity: 0.5
		},
		chipToday: {
			right: 10,
			top: 10,
			position: 'absolute',
			borderWidth: 1,
			borderColor: theme.colors.primary,
			backgroundColor: theme.colors.primaryContainer,
			borderRadius: 50
		},

		chipUpcoming: {
			right: 10,
			top: 10,
			position: 'absolute',
			borderWidth: 1,
			borderColor: theme.colors.error,
			backgroundColor: theme.colors.errorContainer,
			borderRadius: 50
		},
		chipTodayText: {
			color: theme.colors.primary
		},
		chipUpcomingText: {
			color: theme.colors.error
		},
		eventInfo: {
			marginVertical: 10,
			gap: 10
		}
	})

export default EventCard
