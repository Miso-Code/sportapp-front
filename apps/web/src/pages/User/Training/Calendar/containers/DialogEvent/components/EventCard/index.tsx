import { Box, Card, CardContent, Typography } from '@mui/material'
import { format } from 'date-fns'
import { es } from 'date-fns/locale'
import { CustomEvent } from '../../../CustomCalendar/interfaces'

export default function EventCard({
	selectedValue
}: {
	selectedValue: CustomEvent
}) {
	return (
		<Card>
			<CardContent>
				<Box
					sx={{
						display: 'flex',
						justifyContent: 'space-between',
						pb: 2
					}}>
					<Typography variant='body1' color='GrayText'>
						{format(
							selectedValue.start as Date,
							'dd MMMM yyyy hh:mm a',
							{
								locale: es
							}
						)}
					</Typography>
				</Box>
				<Typography gutterBottom variant='h6'>
					{selectedValue.title}
				</Typography>
				<Typography variant='body2' color='GrayText' component='div'>
					{selectedValue.description}
				</Typography>
			</CardContent>
		</Card>
	)
}
