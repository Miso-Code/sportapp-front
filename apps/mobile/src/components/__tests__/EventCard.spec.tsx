import React from 'react'
import renderer, { ReactTestRenderer } from 'react-test-renderer'

import EventCard from '../EventCard'

jest.mock('dayjs')

describe('EventCard', () => {
	let component: ReactTestRenderer
	let spy: jest.SpyInstance

	beforeEach(() => {
		const mockedDate = new Date(1996, 6, 19)

		spy = jest.spyOn(global, 'Date').mockImplementation(() => mockedDate)
		component = renderer.create(
			<EventCard
				date={new Date().toISOString()}
				title='title'
				description='description'
			/>
		)
	})

	afterEach(() => {
		spy.mockRestore()
		component.unmount()
	})

	it('should render correctly', () => {
		expect(component.toJSON()).toMatchSnapshot()
	})

	it('should render today chip', () => {
		expect(
			component.root.findByProps({ testID: 'chip' }).props.children
		).toBe('training.today')
	})

	it('should change the chip on a future date', () => {
		const future = new Date()
		future.setFullYear(future.getFullYear() + 1)
		component.update(
			<EventCard
				date={future.toISOString()}
				title='title'
				description='description'
			/>
		)

		expect(
			component.root.findByProps({ testID: 'chip' }).props.children
		).toBe('training.today')
	})
})
