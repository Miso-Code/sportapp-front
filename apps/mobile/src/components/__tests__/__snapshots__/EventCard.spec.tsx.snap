// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventCard should render correctly 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "#f2f2f2",
      "borderRadius": 12,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 3,
    }
  }
  testID="card-container-outer-layer"
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#f2f2f2",
        "borderRadius": 12,
        "elevation": 1,
        "flex": undefined,
        "paddingBottom": 10,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 1,
      }
    }
    testID="card-container"
  >
    <View
      style={
        [
          {
            "flexShrink": 1,
          },
          undefined,
        ]
      }
      testID="card"
    >
      <View
        borderRadiusStyles={{}}
        style={
          [
            {
              "paddingHorizontal": 16,
            },
            {
              "paddingVertical": 16,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              [
                {
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
                {
                  "opacity": 0.5,
                },
              ],
            ]
          }
        >
          1996-07-19
        </Text>
        <View
          style={
            {
              "gap": 10,
              "marginVertical": 10,
            }
          }
        >
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "500",
                    "letterSpacing": 0.15,
                    "lineHeight": 24,
                  },
                  undefined,
                ],
              ]
            }
          >
            title
          </Text>
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "400",
                    "letterSpacing": 0.25,
                    "lineHeight": 20,
                  },
                  undefined,
                ],
              ]
            }
          >
            description
          </Text>
        </View>
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "rgba(234, 221, 255, 1)",
              "borderRadius": 50,
              "position": "absolute",
              "right": 10,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
              "top": 10,
            }
          }
          testID="chip-container-outer-layer"
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "rgba(234, 221, 255, 1)",
                "borderColor": "rgba(103, 80, 164, 1)",
                "borderRadius": 50,
                "borderStyle": "solid",
                "borderWidth": 1,
                "flex": undefined,
                "flexDirection": "column",
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
            testID="chip-container"
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": true,
                  "expanded": undefined,
                  "selected": false,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "overflow": "hidden",
                  },
                  [
                    {
                      "borderRadius": 50,
                    },
                    {
                      "width": "100%",
                    },
                  ],
                ]
              }
              testID="chip"
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "paddingLeft": 4,
                      "position": "relative",
                    },
                    {
                      "paddingLeft": 0,
                    },
                    {
                      "paddingRight": 0,
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    [
                      {
                        "textAlign": "left",
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "writingDirection": "ltr",
                      },
                      [
                        {
                          "fontFamily": "System",
                          "fontSize": 14,
                          "fontWeight": "500",
                          "letterSpacing": 0.1,
                          "lineHeight": 20,
                        },
                        [
                          {
                            "marginVertical": 6,
                            "textAlignVertical": "center",
                          },
                          {
                            "color": "rgba(29, 25, 43, 1)",
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          {
                            "marginLeft": 12,
                            "marginRight": 12,
                          },
                          {
                            "color": "rgba(103, 80, 164, 1)",
                          },
                        ],
                      ],
                    ]
                  }
                >
                  training.today
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
